pyteamcity/future/teamcity.py

```diff
@@ -79,3 +80,55 @@ def from_environ(cls):
             username=os.environ.get('TEAMCITY_USER'),
             password=os.environ.get('TEAMCITY_PASSWORD'),
             server=os.environ.get('TEAMCITY_HOST'))
+
+    @property
+    def server_info(self):
+        url = self.base_url + '/app/rest/server'
+        res = self.session.get(url)
+        res.raise_for_status()
```

Comment:
You already have `HTTPError`, don't you? You can use this as your base exception for all http exceptions. Later, if you find yourself needing to expose more specific http errors then you can create new exceptions that inherit `HTTPError` -- this won't be backwards incompatible for existing user code.